_test/shared/noisify_test_rand.m

```diff
@@ -0,0 +1,65 @@
+classdef noisify_test_rand < handle
+%--------------------------------------------------
+% This class creates a simple deterministic series
+% of numbers which can be substituted for a true pseudo-random
+% number generator to enable by-eye checking of the random
+% noise addition (via noisify( to the signal. It can
+% substitute for randn in the Herbert noisify method.
+%
+% This function is designed to accumulate the "pseudo-
+% random" numbers across multiple calls - one for each
+% page of a paged sqw object. An external global flag
+% initialises these numbers, and an internal persistent
+% number keeps track of where this process has got to.
+%
+% Input to the generator function myrand:
+%     n - size of the signal vector for which the noise
+%         is being generated.
+% Where multiple tests are required a separate class object
+% can be used for each one.
+%--------------------------------------------------
+
+    properties
+        % flag to signal that this pseudo-random
+        % number distribution needs to be initialised if true.
+        init = 1
+        % number tracking where in the generation of
+        % the sequence we have got to. Initially a dummy bad value.
+        got_to = -666
+    end
+    methods
+        function vals = myrand(obj,size)
+            % On initialisation the flag is reset false and the number
+            % sequence tracker is initialised to 1
+            if obj.init==1
+                obj.init=0;
+                obj.got_to=1;
+            end
+            
+            % the size vector of the signal which is passed in as the argument
+            % is a 1xsize vector - the actual size is extracted from the second 
+            % element.
+            size=size(2);
+            
+            % The values generated are an ascending sequence of increment one in 
```

Comment:
this comment described what the function does and expected input so belongs in the function header rather than the code.

Reply:
Sometimes it is useful for the maintainer to understand what is going on with the particular line of code. This is the implementation. If it went in the header it would be describing the output, the expected interface. It would also be good to indicate whether your comment is saying that something is wrong - doesn't do what is expected - vs does not adhere to some protocol.

Reply:
It's a stylistic thing rather than a right/wrong thing.

The function name is a short summary of what the function does (get_x, add_values, ...) and argument names hints to what the expected input/output arguments should be.

The function header is where a user would look for information about input and output arguments and detail about what the function will do. There shouldn't be need for anyone to look at implementation of functions to understand how to call them or what the effect will be.

The code comment is, as you say, information for the developer on how that is achieved if it's not obvious from the code. or why it's implemented in that way.

Reply:
My favourite book is Conway's "Perl best practices". A lot of it could be summarized as "actually use Python", but apart from that... it has useful comments like "you can  restrict doing [elaborate comment method] to code of more than N lines long .... as long as N=0". Here "if it's not obvious" can be expanded with "(and it will be not obvious)". Developers often tend to start reading the code directly, and only give up and look at the header if it becomes too boggy. Especially true if you are in the debugger where you do start at the first executable line.